sentry-spring/src/main/java/io/sentry/spring/SentrySpringFilter.java

```diff
@@ -0,0 +1,100 @@
+package io.sentry.spring;
+
+import static io.sentry.SentryOptions.RequestSize.*;
+
+import com.jakewharton.nopen.annotation.Open;
+import io.sentry.Breadcrumb;
+import io.sentry.HubAdapter;
+import io.sentry.IHub;
+import io.sentry.SentryLevel;
+import io.sentry.SentryOptions.RequestSize;
+import io.sentry.util.Objects;
+import java.io.IOException;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.http.MediaType;
+import org.springframework.util.MimeType;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@Open
+public class SentrySpringFilter extends OncePerRequestFilter {
+  private final @NotNull IHub hub;
+  private final @NotNull SentryRequestResolver requestResolver;
+
+  public SentrySpringFilter(@NotNull IHub hub, @NotNull SentryRequestResolver requestResolver) {
+    this.hub = Objects.requireNonNull(hub, "hub is required");
+    this.requestResolver = Objects.requireNonNull(requestResolver, "requestResolver is required");
+  }
+
+  public SentrySpringFilter(final @NotNull IHub hub) {
+    this(hub, new SentryRequestResolver(hub));
+  }
+
+  public SentrySpringFilter() {
+    this(HubAdapter.getInstance());
+  }
+
+  @Override
+  protected void doFilterInternal(
+      final @NotNull HttpServletRequest servletRequest,
+      final @NotNull HttpServletResponse response,
+      final @NotNull FilterChain filterChain)
+      throws ServletException, IOException {
+    if (hub.isEnabled()) {
+      final HttpServletRequest request = resolveHttpServletRequest(servletRequest);
+      try {
+        hub.pushScope();
```

Comment:
We're popping inside of the `finally` block so technically this `pushScope` should be before try:


```suggestion
      hub.pushScope();
      try {
```

Reply:
👍